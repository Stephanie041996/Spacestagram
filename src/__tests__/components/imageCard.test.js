import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import  ImageCard from '../../components/imageCard';
import store from '../../Store';

afterEach(cleanup);

it('should render the image Card', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ImageCard />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <ImageCard />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
