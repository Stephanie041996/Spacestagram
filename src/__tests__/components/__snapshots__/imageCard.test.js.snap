// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the image Card 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="mySpan"
  >
    <div
      class="card-header"
    >
      <div
        class="card-title-group"
      >
        <h5
          class="card-title"
        />
        <div
          class="card-date"
        />
      </div>
    </div>
    <img
      alt="LostImage"
      class="card-image"
    />
    <div
      class="card-text"
    />
    <div
      class="card-like-bar"
    >
      <div
        class="card-like-bar"
      >
        <button
          class="card-like-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart fa-w-16 "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="like-text"
        >
          <b>
            Likes: 0
          </b>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render the navbar 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="mySpan"
  >
    <div
      class="card-header"
    >
      <div
        class="card-title-group"
      >
        <h5
          class="card-title"
        />
        <div
          class="card-date"
        />
      </div>
    </div>
    <img
      alt="LostImage"
      class="card-image"
    />
    <div
      class="card-text"
    />
    <div
      class="card-like-bar"
    >
      <div
        class="card-like-bar"
      >
        <button
          class="card-like-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart fa-w-16 "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="like-text"
        >
          <b>
            Likes: 0
          </b>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
